test/d_do_test.d

```diff
@@ -715,6 +747,17 @@ int tryMain(string[] args)
         }
     }
 
+    if(!autoCompileImports && testArgs.compiledImports.length > 0) {
+       autoCompileImports = true;
+       goto AGAIN;
+    }
+
+    argSetIndex++;
+    if(argSetIndex < testArgs.argSets.length) {
+        argSet = testArgs.argSets[argSetIndex];
+        goto AGAIN;
+    }
```

Comment:
Is it really so bad to factor out the loop and do sth. like:

```d
foreach (arg; choice(!tesArgs.empty, testArgs, [""]))
   testCombinations(args, <otherConfigState>);
```

Reply:
I would be fine with that but would require moving the test code to a new function and I didn't want to mess up the diff that much.  Maybe I should do that refactor first in a seperate PR and then rebase this PR off that?

Reply:
Well, we have been working on removes `goto`s from the codebase, so it feels a bit wrong to introduce them if there's no need (of course, sometime they are great).

>  Maybe I should do that refactor first in a seperate PR and then rebase this PR off that?

Yeah that would be awesome!